lib/rules/enzyme/no-bad-features.js

```diff
@@ -0,0 +1,129 @@
+const BAD_FEATURES = {
+  immutability: ['setState', 'setContext', 'simulate', 'simulateError'],
+  'component-boundaries': ['instance', 'state', 'ref'],
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Restricts usage of enzyme to the good parts.',
+      category: 'Best Practices',
+      recommended: true,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/enzyme/no-bad-features.md',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allow: {
+            type: 'array',
+            items: {
+              enum: Object.keys(BAD_FEATURES).reduce((acc, value) => {
+                return [...acc, value, ...BAD_FEATURES[value]];
+              }, []),
+            },
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+
+  create(context) {
+    const enzymeWrappers = [];
+    const options = context.options[0] || {};
+    const allowed = options.allow || [];
+
+    function report(node, prop) {
+      context.report({
+        node,
+        message:
+          'No bad enzyme features. Avoid using {{prop}} on an enzyme wrapper.',
```

Comment:
I think we might need or-feature recommendations in place of “don’t do bad”

Reply:
Yeah good call.